client/my-sites/checkout/composite-checkout/lib/paypal-express-processor.ts

```diff
@@ -27,28 +27,32 @@ export default async function payPalProcessor(
 		siteId,
 		siteSlug,
 		contactDetails,
-		customizedCancelUrl,
 	} = transactionOptions;
 	reduxDispatch( recordTransactionBeginAnalytics( { paymentMethodId: 'paypal' } ) );
 
 	const thankYouUrl = getThankYouUrl();
 	let currentUrl;
 	let currentBaseUrl;
+	let currentUrlQuery;
 	try {
 		currentUrl = window.location.href;
 		currentBaseUrl = window.location.origin;
+		currentUrlQuery = window.location.search;
 	} catch ( error ) {
 		currentUrl = `https://wordpress.com/checkout/${ siteSlug }`;
 		currentBaseUrl = 'https://wordpress.com';
+		currentUrlQuery = '';
 	}
 	const currentUrlWithoutQuery = currentUrl.split( /\?|#/ )[ 0 ];
+	const updatedQuery = new URLSearchParams( currentUrlQuery );
 	const successUrl = thankYouUrl.startsWith( 'http' ) ? thankYouUrl : currentBaseUrl + thankYouUrl;
-	let cancelUrl = customizedCancelUrl;
-	if ( ! cancelUrl ) {
-		cancelUrl = createUserAndSiteBeforeTransaction
-			? currentUrlWithoutQuery + '?cart=no-user'
-			: currentUrlWithoutQuery;
+	if ( createUserAndSiteBeforeTransaction ) {
+		updatedQuery.append( 'cart', 'no-user' );
```

Comment:
It's probable that this added query param is not necessary after the rest of this fix. `cart=no-user` seems to be the main cause of `createUserAndSiteBeforeTransaction`:

https://github.com/Automattic/wp-calypso/blob/c500a3aaca6d06f3c5fd96e27a04fc6da1db329e/client/my-sites/checkout/controller.jsx#L140

https://github.com/Automattic/wp-calypso/blob/c500a3aaca6d06f3c5fd96e27a04fc6da1db329e/client/my-sites/checkout/composite-checkout/composite-checkout.tsx#L151-L152

...and it seems unlikely that that variable is intended for any URL except checkout.

The full logic to get `createUserAndSiteBeforeTransaction` is a little tricky to follow but basically the code says: if `cart=no-user` is in the query string, and the site is not a jetpack checkout site, or if there is not a logged-in user and the site is not a jetpack checkout site, then append `cart=no-user` to the query string.

But I wonder if appending `cart=no-user` is actually important for a logged-out checkout?

However, I'm not 100% certain that it's safe to remove and it should be harmless to duplicate so unless I can determine otherwise I plan to leave it for now.